src/PointwiseFunctions/AnalyticData/NewtonianEuler/KhInstability.hpp

```diff
@@ -0,0 +1,245 @@
+
+namespace NewtonianEuler {
+namespace AnalyticData {
+
+/*!
+ * \brief Initial data to simulate the Kelvin-Helmholtz instability.
+ *
+ * For comparison purposes, this class implements the planar shear of
+ * \cite Schaal2015 The initial state consists of a horizontal strip of mass
+ * density \f$\rho_\text{in}\f$ moving with horizontal speed
+ * \f$v_{\text{in}}\f$. The rest of the fluid possesses mass density
+ * \f$\rho_\text{out}\f$, and its horizontal velocity is \f$v_{\text{out}}\f$,
+ * both constant. Mathematically,
+ *
+ * \f{align*}
+ * \rho(x, y) =
+ * \begin{cases}
+ * \rho_\text{in}, & \left|y - y_\text{mid}\right| < b/2\\
+ * \rho_\text{out}, & \text{otherwise},
+ * \end{cases}
+ * \f}
+ *
+ * and
+ *
+ * \f{align*}
+ * v_x(x, y) =
+ * \begin{cases}
+ * v_{\text{in}}, & \left|y - y_\text{mid}\right| < b/2\\
+ * v_{\text{out}}, & \text{otherwise},
+ * \end{cases}
+ * \f}
+ *
+ * where \f$b > 0\f$ is the thickness of the strip, and \f$y = y_\text{mid}\f$
+ * is its horizontal bimedian. The initial pressure is set equal to a constant,
+ * and the system is evolved assuming an ideal fluid of known adiabatic index.
+ * Finally, in order to excite the instability, the vertical velocity is
+ * initialized to a small perturbation
+ *
+ * \f{align*}
+ * v_y(x, y) = A\sin(4\pi x)
+ * \left[\exp\left(-\dfrac{(y - y_\text{top})^2}{2\sigma^2}\right) +
+ * \exp\left(-\dfrac{(y - y_\text{bot})^2}{2\sigma^2}\right)\right],
+ * \f}
+ *
+ * where \f$A\f$ is the amplitude, \f$\sigma\f$ is a damping factor, and
+ * \f$y_\text{top} = y_\text{mid} + b/2\f$ and
+ * \f$y_\text{bot} = y_\text{mid} - b/2\f$ are the vertical coordinates
+ * of the top and bottom boundaries of the strip, respectively.
+ *
+ * \note This class can be used to initialize 2D and 3D data. In 3D, the strip
+ * is aligned to the \f$x-\f$axis, and the vertical direction is taken to be
+ * the \f$z-\f$axis.
+ */
+template <size_t Dim>
+class KhInstability : public MarkAsAnalyticData {
+ public:
+  using equation_of_state_type = EquationsOfState::IdealFluid<false>;
+  using source_term_type = Sources::NoSource;
+
+  /// The adiabatic index of the fluid.
+  struct AdiabaticIndex {
+    using type = double;
+    static constexpr OptionString help = {"The adiabatic index of the fluid."};
+  };
+
+  /// The vertical coordinate of the horizontal bimedian of the strip.
+  struct StripBimedian {
+    using type = double;
+    static constexpr OptionString help = {"The height of the strip center."};
+  };
+
+  /// The thickness of the strip.
+  struct StripThickness {
+    using type = double;
+    static type lower_bound() noexcept { return 0.0; }
+    static constexpr OptionString help = {
+        "The thickness of the horizontal strip."};
+  };
+
+  /// The mass density in the strip
+  struct StripDensity {
+    using type = double;
+    static type lower_bound() noexcept { return 0.0; }
+    static constexpr OptionString help = {
+        "The mass density below the interface"};
```

Comment:
Here and below, the help strings talk about "below the interface" and "above the interface", but the documentation makes no mention of interface and/or what it means to be above/below it. I would say "in the strip" and "outside of the strip" instead... this would better match the rho_in, rho_out in the documentation.